lib/thread.c

```diff
@@ -1511,9 +1511,12 @@ struct thread *thread_fetch(struct thread_master *m, struct thread *fetch)
 			if (fetch->ref)
 				*fetch->ref = NULL;
 			pthread_mutex_unlock(&m->mtx);
+			GETRUSAGE(&m->last_getrusage);
```

Comment:
It looks like this just moves the GETRUSAGE() call out from thread_call() in this path - it's still called once here for every ready task? maybe you could test the boolean to decide whether you need to call GETRUSAGE(), and only call it here the the first time, when "ready_run" is "false"?